### DISCUSSION TITLE
GeoIP is too basic

### DISCUSSION THREAD

**Discussion by LucidAPs** [Zrips/CMI#6051] [Enhancement]
The GeoIP detection CMI offers is not accurate at all and generally very lackluster, especially compared to EssentialsX MaxMind GeoLite2 database lookups.

**Comment by mrfloris** [comment]
Yeah, this can probably use an update, it's been years since it got new info.

**Comment by spannerman79** [comment]
To properly update this the databases must be downloaded a little differently these days.

https://dev.maxmind.com/geoip/geolite2-free-geolocation-data?lang=en

To automate the updated DB's you use;

GeoLite2 Country

```
# Database URL
https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-Country&license_key=YOUR_LICENSE_KEY&suffix=tar.gz

# SHA256 URL
https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-Country&license_key=YOUR_LICENSE_KEY&suffix=tar.gz.sha256
```

GeoLite2 City

```
# Database URL
https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=YOUR_LICENSE_KEY&suffix=tar.gz

# SHA256 URL
https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City&license_key=YOUR_LICENSE_KEY&suffix=tar.gz.sha256
```

For both replace `YOUR_LICENSE_KEY` with the key you generate via https://www.maxmind.com/en/accounts/current/license-key?lang=en

https://maxmind.github.io/GeoIP2-java/
